### ISSUE TITLE
Validation returns "An operation is not implemented: something went wrong" for complex schema

### ISSUE THREAD

**Issue by rayokota** [erosb/json-sKema#26]
Hi @erosb ,

While using this library against our test suite, I ran into a validation error on a complex schema.  Below is the code that returns the error.

```
package io.confluent.kafka.schemaregistry.json.schema;

import com.github.erosb.jsonsKema.IJsonValue;
import com.github.erosb.jsonsKema.JsonParseException;
import com.github.erosb.jsonsKema.JsonParser;
import com.github.erosb.jsonsKema.JsonValue;
import com.github.erosb.jsonsKema.Schema;
import com.github.erosb.jsonsKema.SchemaClient;
import com.github.erosb.jsonsKema.SchemaLoader;
import com.github.erosb.jsonsKema.SchemaLoaderConfig;
import com.github.erosb.jsonsKema.SchemaLoadingException;
import com.github.erosb.jsonsKema.Validator;
import java.io.BufferedReader;
import java.io.ByteArrayInputStream;
import java.io.FileNotFoundException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.UncheckedIOException;
import java.net.URI;
import java.nio.charset.StandardCharsets;
import java.util.HashMap;
import java.util.Map;
import java.util.stream.Collectors;
import org.junit.Test;

public class ReferenceTest {

  private static final String DUMMY_BASE_URI = "dummy://input/";

  @Test
  public void testReferences() throws Exception {
    String doc = "{\n"
        + "  \"customerName\": \"acme\",\n"
        + "  \"acquireDate\": \"2020-12-12\"\n"
        + "}";
    JsonValue jsonValue = new JsonParser(doc).parse();

    String oneOfSchema = "{\n"
        + "  \"oneOf\": [\n"
        + "    { \"$ref\": \"customer.json\"},\n"
        + "    { \"$ref\": \"user.json\"}\n"
        + "  ]\n"
        + "}";
    String userSchema = "{\n"
        + "  \"type\": \"object\",\n"
        + "  \"additionalProperties\": false,\n"
        + "  \"properties\": {\n"
        + "    \"name\": {\n"
        + "      \"type\": \"string\"\n"
        + "    },\n"
        + "    \"age\": {\n"
        + "      \"type\": \"integer\",\n"
        + "      \"minimum\": 0\n"
        + "    }\n"
        + "  },\n"
        + "  \"required\": [\n"
        + "    \"age\"\n"
        + "  ]\n"
        + "}\n";
    String customerSchema = "{\n"
        + "  \"type\": \"object\",\n"
        + "  \"additionalProperties\": false,\n"
        + "  \"properties\": {\n"
        + "    \"customerName\": {\n"
        + "      \"type\": \"string\"\n"
        + "    },\n"
        + "    \"acquireDate\": {\n"
        + "      \"type\": \"string\"\n"
        + "    }\n"
        + "  }\n"
        + "}\n";

    Map<URI, String> refs = new HashMap<>();
    refs.put(new URI(DUMMY_BASE_URI + "user.json"), userSchema);
    refs.put(new URI(DUMMY_BASE_URI + "customer.json"), customerSchema);

    SchemaLoaderConfig config = new SchemaLoaderConfig(
        new ReferenceSchemaClient(refs), DUMMY_BASE_URI);

    JsonValue schemaJson = new JsonParser(oneOfSchema).parse();

    Schema loadedSchema = new SchemaLoader(schemaJson, config).load();
    Validator validator = Validator.forSchema(loadedSchema);
    validator.validate(jsonValue);
  }

  public static class ReferenceSchemaClient implements SchemaClient {

    private Map<URI, String> references;

    public ReferenceSchemaClient(Map<URI, String> references) {
      this.references = references;
    }

    @Override
    public InputStream get(URI uri) {
      String reference = references.get(uri);
      if (reference == null) {
        throw new UncheckedIOException(new FileNotFoundException(uri.toString()));
      }
      return new ByteArrayInputStream(reference.getBytes(StandardCharsets.UTF_8));
    }

    @Override
    public IJsonValue getParsed(URI uri) {
      BufferedReader reader = new BufferedReader(new InputStreamReader(get(uri)));
      String string = reader.lines().collect(Collectors.joining());
      try {
        return new JsonParser(string, uri).parse();
      } catch (JsonParseException ex) {
        throw new SchemaLoadingException("failed to parse json content returned from $uri", ex);
      }
    }
  }
}
```

Oddly, if I remove `"required": ["age"]` from `userSchema`, then the "something went wrong" error does not appear.

**Comment by rayokota** [comment]
Here is the full stack trace 

```
kotlin.NotImplementedError: An operation is not implemented: something went wrong

	at com.github.erosb.jsonsKema.AggregatingValidationFailure.join$json_sKema(ValidationFailure.kt:67)
	at com.github.erosb.jsonsKema.DefaultValidator.accumulate(Validator.kt:617)
	at com.github.erosb.jsonsKema.DefaultValidator.accumulate(Validator.kt:95)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitChildren(SchemaVisitor.kt:122)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitCompositeSchema(SchemaVisitor.kt:59)
	at com.github.erosb.jsonsKema.DefaultValidator.visitCompositeSchema(Validator.kt:186)
	at com.github.erosb.jsonsKema.DefaultValidator.visitCompositeSchema(Validator.kt:95)
	at com.github.erosb.jsonsKema.SchemaVisitor.internallyVisitCompositeSchema$json_sKema(SchemaVisitor.kt:48)
	at com.github.erosb.jsonsKema.CompositeSchema.accept(Schema.kt:26)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitChildren(SchemaVisitor.kt:121)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitReferenceSchema(SchemaVisitor.kt:84)
	at com.github.erosb.jsonsKema.ReferenceSchema.accept(Schema.kt:32)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitChildren(SchemaVisitor.kt:121)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitCompositeSchema(SchemaVisitor.kt:59)
	at com.github.erosb.jsonsKema.DefaultValidator.visitCompositeSchema(Validator.kt:186)
	at com.github.erosb.jsonsKema.DefaultValidator.visitCompositeSchema(Validator.kt:95)
	at com.github.erosb.jsonsKema.SchemaVisitor.internallyVisitCompositeSchema$json_sKema(SchemaVisitor.kt:48)
	at com.github.erosb.jsonsKema.CompositeSchema.accept(Schema.kt:26)
	at com.github.erosb.jsonsKema.DefaultValidator.visitOneOfSchema(Validator.kt:516)
	at com.github.erosb.jsonsKema.DefaultValidator.visitOneOfSchema(Validator.kt:95)
	at com.github.erosb.jsonsKema.OneOfSchema.accept(OneOf.kt:7)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitChildren(SchemaVisitor.kt:121)
	at com.github.erosb.jsonsKema.SchemaVisitor.visitCompositeSchema(SchemaVisitor.kt:59)
	at com.github.erosb.jsonsKema.DefaultValidator.visitCompositeSchema(Validator.kt:186)
	at com.github.erosb.jsonsKema.DefaultValidator.visitCompositeSchema(Validator.kt:95)
	at com.github.erosb.jsonsKema.SchemaVisitor.internallyVisitCompositeSchema$json_sKema(SchemaVisitor.kt:48)
	at com.github.erosb.jsonsKema.CompositeSchema.accept(Schema.kt:26)
	at com.github.erosb.jsonsKema.DefaultValidator.validate(Validator.kt:173)
	at io.confluent.kafka.schemaregistry.json.schema.ReferenceTest.testReferences(ReferenceTest.java:84)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)

```

**Comment by erosb** [comment]
@rayokota I published a new version `0.9.0` with the recent fixes (should be available on maven central in a couple of hours).

**Comment by rayokota** [comment]
Thanks @erosb !  This helps a lot :)
